lxd/instance/drivers/driver_common.go

```diff
@@ -822,6 +822,17 @@ func (d *common) snapshotCommon(inst instance.Instance, name string, expiry *tim
 		return err
 	}
 
+	if pool.Driver().Info().RunningCopyFreeze && inst.IsRunning() && !inst.IsFrozen() {
+		// Freeze the processes.
+		err = inst.Freeze()
+		if err != nil {
+			return err
+		}
+
+		defer func() { _ = inst.Unfreeze() }()
```

Comment:
please get into the habit of logging warnings when errors occur in defer/reverts.